crates/extrinsics/src/instantiate.rs

```diff
@@ -363,13 +368,17 @@ impl InstantiateExec {
         code: Vec<u8>,
         gas_limit: Weight,
     ) -> Result<InstantiateExecResult, ErrorVariant> {
-        let call = api::tx().contracts().instantiate_with_code(
-            self.args.value,
-            gas_limit.into(),
-            self.args.storage_deposit_limit_compact(),
-            code.to_vec(),
-            self.args.data.clone(),
-            self.args.salt.clone(),
+        let call = subxt::tx::Payload::new(
+            "Contracts",
+            "instantiate_with_code",
+            InstantiateWithCode {
+                value: self.args.value,
+                gas_limit: gas_limit.into(),
+                storage_deposit_limit: self.args.storage_deposit_limit_compact(),
+                code,
+                data: self.args.data.clone(),
+                salt: self.args.salt.clone(),
+            },
```

Comment:
Using the metadata generated API gave us an error when attempting to interact with a version of `pallet_contracts` with incompatible metadata. This was annoying but useful.

We should consider adding some safety and UX back by at least checking that the constructed call we have is compatible with the target node. It will prevent either submitting an extrinsic which succeeds to decode into a different form (bad), or less worse the user just getting a not useful error because the extrinsic could not be decoded.

Reply:
Okay great I didn't know about that!